[NG] fix(signpost): Adds a11y to the signpost

hippee-lee opened the pull request:
- Adds proper aria attributes to signpost trigger and signpost content components
- adds service for uniq id's (on content and used in aria-controls
- closes #3523

Signed-off-by: Matt Hippely <mhippely@vmware.com>

coryrylan commented on src/clr-angular/popover/signpost/_signposts.clarity.scss:
```diff
@@ -340,20 +340,25 @@
 
   .signpost-content-body {
-    flex: 1 1 auto;
+    //flex: 1 1 auto;
```
nit: delete the unused commented code

hippee-lee replied:
I removed this and two others

coryrylan commented on src/clr-angular/popover/signpost/signpost.spec.ts:
```diff
@@ -97,6 +103,50 @@ export default function(): void {
+      function checkAriaControlsId(id: string, element: HTMLElement) {
+        const triggerControlsValue = element.querySelector('.signpost-action').getAttribute('aria-controls');
+        const closeControlsValue = element
+          .querySelector('.signpost-content > .signpost-wrap >' + ' .signpost-content-header > .signpost-action')
```
Could this querySelector be simplified? Seems like it doesn't need to append two separate strings.

hippee-lee replied:
I simplified this.

hippee-lee commented:
While addressing feedback I updated the reference images. Due to the layout changes to make the projected content accessible there were small visual changes to the dev app content and focus behavior.

mathisscott commented on src/clr-angular/popover/signpost/providers/signpost-id.service.ts:
```diff
@@ -0,0 +1,23 @@
+  private _id: Subject<string> = new Subject<string>();
+
+  setId(id: string) {
+    this._id.next(id);
```
Why is this `setId()` and not `set id():`?

hippee-lee replied:
Getters and setters in typescript need to have the same type. I'm using a getter for the Observable to have cleaner syntax when subscribing and that means I can't have  a setter with a boolean arg.

mathisscott commented on src/clr-angular/popover/signpost/signpost-trigger.ts:
```diff
@@ -19,16 +30,30 @@ import { IfOpenService } from '../../utils/conditional/if-open.service';
  */
 export class ClrSignpostTrigger implements OnDestroy {
   private subscriptions: Subscription[] = [];
+  public ariaExpanded;
```
Does `ariaExpanded` have a type?

It looks like it's a boolean...

hippee-lee replied:
I thought it would get inferred by the `tsc` but will add it for Clarity.

mathisscott commented on src/clr-angular/popover/signpost/signpost.spec.ts:
```diff
@@ -97,6 +103,50 @@ export default function(): void {
+          .getAttribute('aria-controls');
+        const contentId = element.querySelector('.signpost-content').getAttribute('id');
+
+        expect(id).toBe(contentId);
```
This is great but is it clear where this breaks when it breaks?

If not, we may want to add the optional string on the expect...

`expect(id).toBe(whatWeAreTesting, 'string that is passed that describes what we are testing and why it might be broken')`

hippee-lee replied:
I will add the string to the expect for more descriptive failures.
